cpp/src/arrow/array-binary-test.cc

```diff
@@ -184,7 +185,7 @@ TEST_F(TestStringArray, TestSliceGetString) {
   std::shared_ptr<Array> array;
   ASSERT_OK(builder.Finish(&array));
   auto s = array->Slice(1, 10);
-  auto arr = std::dynamic_pointer_cast<StringArray>(s);
+  auto arr = checked_pointer_cast<StringArray>(s);
```

Comment:
In tests, I think we would like to keep `dynamic_pointer_cast` (e.g. if tests are built in release mode)

Reply:
This patch came out of the Fedora compiler problems, reverted it.

Reply:
I don't think we should aim at making Arrow work without RTTI.